File: be/ClinicManager/src/main/java/com/clinic/clinicmanager/model/Review.java
package com.clinic.clinicmanager.model;

import com.clinic.clinicmanager.model.constants.ReviewSource;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "review")
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "client_id", nullable = false)
    private Client client;

    @Column(nullable = false)
    @Builder.Default
    private Boolean isUsed = false;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReviewSource source;

    @Column(nullable = false)
    private LocalDate issueDate;

    @Column(name = "created_by_user_id")
    private Long createdByUserId;
}

File: fe/ClinicManager/src/models/review.ts
import { Client } from "./client";

export enum ReviewSource {
    BOOKSY = "BOOKSY",
    GOOGLE = "GOOGLE"
}

export interface Review {
    id: string | number,
    client: Client,
    isUsed: boolean,
    source: ReviewSource,
    issueDate: string,
    createdBy: number | null;
}

export interface NewReview {
    client: Client | null,
    isUsed?: boolean | null,
    source: ReviewSource | null,
    issueDate: string,
}

export interface ReviewFilterDTO {
    keyword?: string | null,
    source?: ReviewSource | null,
    isUsed?: Boolean | null,
}